rust/hyperlane-core/src/types/mod.rs

```diff
@@ -139,3 +140,13 @@ impl TxCostEstimate {
         self.l2_gas_limit.unwrap_or(self.gas_limit)
     }
 }
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub enum IsmType {
```

Comment:
You can rm this type and use `ModuleType` in `rust/hyperlane-core/src/traits/interchain_security_module.rs` instead. But probably need to keep the borsh derivations